src/query/graphite/native/builtin_functions.go

```diff
@@ -1658,6 +1658,83 @@ func movingMedian(ctx *common.Context, _ singlePathSpec, windowSize string) (*bi
 	}, nil
 }
 
+// movingSum takes one metric or a wildcard seriesList followed by a a quoted string
+// with a length of time like '1hour' or '5min'. Graphs the sum of the preceding
+// datapoints for each point on the graph. All previous datapoints are set to None at
+// the beginning of the graph.
+func movingSum(ctx *common.Context, _ singlePathSpec, windowSize string) (*binaryContextShifter, error) {
+	interval, err := common.ParseInterval(windowSize)
+	if err != nil {
+		return nil, err
+	}
+
+	if interval <= 0 {
+		return nil, common.ErrInvalidIntervalFormat
+	}
+
+	contextShiftingFn := func(c *common.Context) *common.Context {
+		opts := common.NewChildContextOptions()
+		opts.AdjustTimeRange(0, 0, interval, 0)
+		childCtx := c.NewChildContext(opts)
+		return childCtx
+	}
+
+	bootstrapStartTime, bootstrapEndTime := ctx.StartTime.Add(-interval), ctx.StartTime
+	transformerFn := func(bootstrapped, original ts.SeriesList) (ts.SeriesList, error) {
+		bootstrapList, err := combineBootstrapWithOriginal(ctx,
+			bootstrapStartTime, bootstrapEndTime,
+			bootstrapped, singlePathSpec(original))
+		if err != nil {
+			return ts.NewSeriesList(), err
+		}
+
+		results := make([]*ts.Series, 0, original.Len())
+
+		for i, bootstrap := range bootstrapList.Values {
+			series := original.Values[i]
+			windowPoints := int(interval / (time.Duration(series.MillisPerStep()) * time.Millisecond))
+			if windowPoints <= 0 {
+				err := errors.NewInvalidParamsError(fmt.Errorf(
+					"non positive window points, windowSize=%s, stepSize=%d",
+					windowSize, series.MillisPerStep()))
+				return ts.NewSeriesList(), err
+			}
+			window := make([]float64, windowPoints)
```

Comment:
nit: Would probably loop through all series before this loop and find the "max window points" required by a series, then reuse that in the inner loop so we don't need to allocate this intermediate large slice per series.

Perhaps we can update movingAverage and movingMedian to do that too?

e.g.
```go
		results := make([]*ts.Series, 0, original.Len())
		maxWindowPoints := 0
		for i, bootstrap := range bootstrapList.Values {
			series := original.Values[i]
			windowPoints := int(interval / (time.Duration(series.MillisPerStep()) * time.Millisecond))
			if windowPoints > maxWindowPoints {
				maxWindowPoints = windowPoints
			}
		}

		windowPoints := make([]float64, maxWindowPoints)
		for i, bootstrap := range bootstrapList.Values {
			series := original.Values[i]
			currWindowPoints := int(interval / (time.Duration(series.MillisPerStep()) * time.Millisecond))
			window := windowPoints[:currWindowPoints]
			util.Memset(window, math.NaN())
			// .. existing code
		}
```